pkg/controller/statusmanager/status_manager.go

```diff
@@ -287,20 +287,34 @@ func (status *StatusManager) writeHypershiftStatus(operStatus *operv1.NetworkSta
 		return
 	}
 	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		hcp := &hyperv1.HostedControlPlane{ObjectMeta: metav1.ObjectMeta{Name: status.hyperShiftConfig.Name}}
-		err := status.client.ClientFor(names.ManagementClusterName).CRClient().Get(
-			context.TODO(), types.NamespacedName{Namespace: status.hyperShiftConfig.Namespace, Name: status.hyperShiftConfig.Name}, hcp)
+
+		hcp := &uns.Unstructured{}
+		hcp.SetGroupVersionKind(hypershift.HostedControlPlaneGVK)
+		err := status.client.ClientFor(names.ManagementClusterName).CRClient().Get(context.TODO(), types.NamespacedName{Namespace: status.hyperShiftConfig.Namespace, Name: status.hyperShiftConfig.Name}, hcp)
 		if err != nil {
-			if errors.IsNotFound(err) {
-				log.Printf("Did not find hostedControlPlane")
-			} else {
-				return err
-			}
+			return err
+		}
+
+		conditionsRaw, _, err := uns.NestedSlice(hcp.UnstructuredContent(), "status", "conditions")
+		if err != nil {
+			return fmt.Errorf("failed extract conditions: %v", err)
 		}
 
-		oldStatus := hcp.Status.DeepCopy()
+		var conditions []metav1.Condition
+		jsonData, err := json.Marshal(conditionsRaw)
+		if err != nil {
+			return fmt.Errorf("error marshalling JSON: %v\n", err)
+		}
+		err = json.Unmarshal(jsonData, &conditions)
+		if err != nil {
+			return fmt.Errorf("Error unmarshalling JSON: %v\n", err)
+		}
+
+		oldConditions := make([]metav1.Condition, len(conditions))
```

Comment:
This whole condition handling could also me moved to the hypershift package, soemhting like maybe `AddConditiontoHostedControlPlane`?